packages/react/src/jsx-namespace.ts

```diff
@@ -54,47 +54,53 @@ type ReactJSXIntrinsicElements = true extends IsPreReact19
   : /** @ts-ignore */
     React.JSX.IntrinsicElements
 
+// based on the code from @types/react@18.2.8
+// https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3197efc097d522c4bf02b94e1a0766d007d6cdeb/types/react/index.d.ts#LL3204C13-L3204C13
+/** @ts-ignore */
```

Comment:
What's the new `@ts-ignore` for? Does the comment get moved around in the false case get moved around when generating the `.d.ts` file?

Reply:
Seems like the answer is the comment gets removed but if that's true, shouldn't the comments be updated for the types above as well? Seems like you "fixed" it for the tests by bumping `@types/react` but that wouldn't fix it for consumers?

Reply:
Maybe keeping this in a `.d.ts` file might make sense so we don't have to rely on comments being preserved in particular places? (don't have strong feelings about that though, just an idea)

Reply:
the one we had was removed, a JSDoc-style comment - in general - should be preserved but perhaps they are only preserved on accessible symbols. In the emitted declaration this gets folded into a single line and the new comment stays in place.

> Seems like you "fixed" it for the tests by bumping @types/react but that wouldn't fix it for consumers?

I was checking the emitted declaration files and this shouldn't affect consumers (for the reason outlined above). That said... now I'm surprised why this was erroring with the new `@types/react` in the first place. I updated the types before fighting this 🤔 

> Maybe keeping this in a .d.ts file might make sense so we don't have to rely on comments being preserved in particular places? (don't have strong feelings about that though, just an idea)

I agree it's not ideal but also not a big deal. The general mantra I've heard (from the TS team members too) is that `.d.ts` shouldn't quite be used when authoring libs in TS. I don't recall the exact reasons behind this though. Testing things out in TS workbench (see [this](https://www.typescriptlang.org/dev/bug-workbench/?out=true&outDir=dist#code/PTAEAEHsFcBcBECWAnAXKAJogzrAsAFAgQYCmAxgDYCGy1sikAdurMtKYYceAGaKVSTagFtS6REwAOcAHSxshRCKmRksUAG9QAMUiRQAX1C9kkEaADks4GSrZLXAqQAeq9aHLNcoAEbVhFl19UABeLRN9dEtLIycefkFhMXQ7SmxZDHlFAkJXdw1YAE8pUmCDcO1eKNBcZEkAcyMgA))... it seems that TS might not even "copy over" dts files into the `distDir`. You can see the output in the Debug tab in the "Virtual File System" section

Reply:
I mean, we could also just keep this `.d.ts` outside of `src`... but ugh, I don't feel overly confident that all of the relative paths would be rewritten correctly. Authoring everything in `src` feels safer/better

Reply:
Continued in https://github.com/emotion-js/emotion/pull/3282